recipes/opengrm/all/conanfile.py

```diff
@@ -0,0 +1,124 @@
+from conans import ConanFile, AutoToolsBuildEnvironment, tools
+from conans.errors import ConanInvalidConfiguration
+import functools
+import os
+
+required_conan_version = ">=1.33.0"
+
+
+class OpenGrmConan(ConanFile):
+    name = "opengrm"
+    description = "The OpenGrm Thrax tools compile grammars expressed as regular expressions and context-dependent rewrite rules into weighted finite-state transducers."
+    topics = ("fst", "wfst", "opengrm", "thrax")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://www.opengrm.org/twiki/bin/view/GRM/Thrax"
+    license = "Apache-2.0"
+
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "enable_bin": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "enable_bin": True,
+        "openfst:enable_grm": True,
```

Comment:
you cannot change default options of other libraries here, it will always fail on missing binaries 🤔

Reply:
so how can I enforce this dependency? without enable_grm, the library cannot be built

Reply:
in `configure` method. Usually we only abort the build in `validate`, but enable_grm is False by default, which means, it need to be forced.

Reply:
So, I need to force that in `configure`? But how can I check the option of dependent package there. You mentioned `validate`, do I also need to add some enforcement there?

Reply:
@npuichigo Yes. Please, take a look here as a real example: https://github.com/conan-io/conan-center-index/blob/master/recipes/kealib/all/conanfile.py

Reply:
Thanks, I have fixed that. But since no `enable_grm` enabled package is available on conancenter, there's still errors.
```
openfst/1.8.2: WARN: Can't find a 'openfst/1.8.2' package for the specified settings, options and dependencies:
- Settings: arch=x86_64, build_type=Release, compiler=gcc, compiler.libcxx=libstdc++11, compiler.version=10, os=Linux
- Options: enable_bin=True, enable_compact_fsts=False, enable_compress=False, enable_const_fsts=False, enable_far=False, enable_grm=True, enable_linear_fsts=False, enable_lookahead_fsts=False, enable_mpdt=False, enable_ngram_fsts=False, enable_pdt=False, enable_special=False, fPIC=True, shared=False
```

So is there any way of conan to build package with specific options on conancenter? Or does all the package hosted in conancenter have to use default options so no other versions are available?

Reply:
please, read the comment below